TASK-52372: XSS in create document form.

IlhemEssaadi opened the pull request:
ISSUES : XSS in create document form: Type a script in the document name when creating a new document, the script is executed and the alert popup displayed.
FIX : the problem fixed by adding a validator to the input text.

houssemexo26 committed:
TASK-35704: Create FB maintenance and update projects versions/dependencies

wassimmhamdi committed:
Task-48942: the comment for the versions is added to the activity of the initial document (copied/pasted) (#1577)

Problem: The comment of the version of the documentY is added in the activity of the documentX
Fix: Version comment of documentY are added to the correct activity of documentY

hakermi committed:
MAINT-36446: Display i18n drives labels in drives selector of the file attachment drawer (#1584)

ISSUE: The drives selector in the file attachment drawer was displaying the english default drives labels
FIX: Get the i18n drive label from the title and display it instead of the default one

hakermi committed:
MAINT-36446: Fix sections labels are not i18n in file attachment drawer (#1594)

ISSUE: In the attachment drawer the section labels are not well translated because of missed keys and unhadnled tranlation for their names
FIX: Add new keys for the section and get the i18n values when display these labels

wassimmhamdi committed:
Task-48942: the comment for the versions is added to the activity of the initial document (copied/pasted) (#1593)

Problem: The comment of the version of the documentY is added in the activity of the documentX
Fix: Version comment of documentY are added to the correct activity of documentY

IlhemEssaadi committed:
TASK-52372: XSS in create document form.

ISSUES : XSS in create document form: Type a script in the document name when creating a new document, the script is executed and the alert popup displayed.
FIX : the problem fixed by adding a validator to the input text.

rdenarie commented on core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/documents/UINewDocumentForm.java:
```diff
@@ -83,10 +86,11 @@
    *
    */
-  public UINewDocumentForm() {
+  public UINewDocumentForm() throws Exception{
```
```suggestion
  public UINewDocumentForm() {
```

rdenarie replied:
Please fix this point

rdenarie commented on core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/documents/UINewDocumentForm.java:
```diff
@@ -83,10 +86,11 @@
     this.documentService = this.getApplicationComponent(DocumentService.class);
     // Title textbox
     UIFormStringInput titleTextBox = new UIFormStringInput(FIELD_TITLE_TEXT_BOX, FIELD_TITLE_TEXT_BOX, null);
+    titleTextBox.addValidator(MandatoryValidator.class).addValidator(ECMNameValidator.class).addValidator(XSSValidator.class);
```
Why did you add ECMNameValidator ?

IlhemEssaadi replied:
> Why did you add ECMNameValidator ?

The addition of the ECMNameValidator is mandatory so that the name of the document is compatible with jcr.

rdenarie replied:
In UINewDocumentForm, the title entered by the user is already updated to fit JCR constraint  :

>       title = getFileName(title, template);

So, no need to add a validator to check it.
The only needed validator is the one for XSS

montassarhamadi committed:
Task-48707: 404 error on documents application page console. (#1599)

Problem: Before this fix, many errors, 404 related to cloud drive stay appeared in the console when the user open doc apps in his space, and those errors indicate that cannot find some paths if  user is not connected to one of provider's cloud drive.
Fix: the fix represented in change the response of DriveService.java from status not found (404) to no content (204) because the error is originally coming from the browser and not the app (browser behavior). On the other side in front, we add a  check on the existence of the object drive which contains all information about user connection to the drive.

IlhemEssaadi committed:
Fix XSS Validator

rdenarie commented on core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/documents/UINewDocumentForm.java:
```diff
@@ -32,6 +32,8 @@
 
 import org.exoplatform.ecm.utils.text.Text;
 import org.exoplatform.ecm.webui.component.explorer.UIJCRExplorer;
+import org.exoplatform.ecm.webui.form.validator.ECMNameValidator;
```
Not necessary
```suggestion
```

rdenarie commented on core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/documents/UINewDocumentForm.java:
```diff
@@ -83,10 +86,11 @@
     this.documentService = this.getApplicationComponent(DocumentService.class);
     // Title textbox
     UIFormStringInput titleTextBox = new UIFormStringInput(FIELD_TITLE_TEXT_BOX, FIELD_TITLE_TEXT_BOX, null);
+    titleTextBox.addValidator(MandatoryValidator.class).addValidator(XSSValidator.class);
```
I think the mandatory validator is not necessary
Before your fix, if you leave the title empty, there is already a message. So please remove this validator (dont forget to remove the import)

rdenarie commented on core/webui/src/main/java/org/exoplatform/ecm/webui/form/validator/XSSValidator.java:
```diff
@@ -19,23 +19,33 @@
     if (inputValue == null || inputValue.trim().length() == 0) {
       return;
     }
+    try{
+      inputValue = URLDecoder.decode(inputValue, StandardCharsets.UTF_8);
+    }catch (Exception e){
+      LOG.warn(e.getMessage());
+    }
 
```
The function already throws Exception, no need to catch and log it

IlhemEssaadi committed:
Fix all the comments requested

rdenarie commented on core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/documents/UINewDocumentForm.java:
```diff
@@ -31,9 +31,7 @@
-import org.exoplatform.ecm.webui.component.explorer.UIJCRExplorer;
-import org.exoplatform.ecm.webui.form.validator.ECMNameValidator;
-import org.exoplatform.ecm.webui.form.validator.XSSValidator;
+import org.exoplatform.ecm.webui.component.explorer.UIJCRExplorer;import org.exoplatform.ecm.webui.form.validator.XSSValidator;
```
Please use the correct formatting

rdenarie commented on core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/documents/UINewDocumentForm.java:
```diff
@@ -136,7 +137,6 @@ public void execute(Event<UINewDocumentForm> event) throws Exception {
       }
       // Get title
       String title = uiDocumentForm.getUIStringInput(FIELD_TITLE_TEXT_BOX).getValue();
-
```
remove this modification

rdenarie commented on core/webui/src/main/java/org/exoplatform/ecm/webui/form/validator/XSSValidator.java:
```diff
@@ -19,24 +19,29 @@
 import org.apache.commons.lang.StringUtils;
 
 import org.exoplatform.commons.utils.HTMLSanitizer;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
```
Not needed

rdenarie commented on core/webui/src/main/java/org/exoplatform/ecm/webui/form/validator/XSSValidator.java:
```diff
@@ -19,24 +19,29 @@
  */
 public class XSSValidator implements Validator {
-
+  private static final Log LOG = ExoLogger.getLogger(XSSValidator.class);
```
not needed

IlhemEssaadi committed:
eliminate unwanted codes

rdenarie commented on core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/documents/UINewDocumentForm.java:
```diff
@@ -29,9 +29,9 @@
 import javax.jcr.nodetype.NodeDefinition;
 
 import org.apache.commons.lang.StringUtils;
-
```
Please remove this modification. We need to keep the minimum modified lines

IlhemEssaadi committed:
remove modification

rdenarie commented on core/webui/src/main/java/org/exoplatform/ecm/webui/form/validator/XSSValidator.java:
```diff
@@ -24,19 +24,21 @@
  * phongth@exoplatform.com Jul 24, 2012
  */
 public class XSSValidator implements Validator {
-
```
Please remove this modification

IlhemEssaadi committed:
Update PR

rdenarie committed:
Merge branch 'feature/maintenance' into TASK-52372
